server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
Although, actually, now that I think about it, does the spec say that it's an error if there is no `targets.json`, so long as there is no entry for one in `snapshots.json`?